src/list.go

```diff
@@ -164,11 +165,61 @@ func ListGenericBackends() ([]string, error) {
 
 	var backends []string
 
+	version, err := GetVaultVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	// With the 0.8.3 release of vault the "generic" backend was renamed to "kv". For backwards
+	// compatibility choose the right one based on the vault version.
+	var backendType string
+	if (version.Minor == 8 && version.Patch == 3) || version.Minor >= 9 || version.Major > 1 {
+		backendType = "kv"
+	} else {
+		backendType = "generic"
```

Comment:
I usually try to avoid else whenever possible (makes code less complex). In this case else is not needed and you could simple assign "generic" to backendType when you initialize it in line 175.